tests/test_fmi3.py

```diff
@@ -828,12 +804,214 @@ def test_directional_derivatives(self):
         assert dv[0] == 1
         assert dv[1] == -5
 
+    def test_get_variable_by_valueref(self):
+        """Test get_variable_by_valueref."""
+        fmu = self._get_reference_fmu("Feedthrough")
+        assert fmu.get_variable_by_valueref(0) == "time"
+        assert fmu.get_variable_by_valueref(1) == "Float32_continuous_input"
+
+    def test_get_variable_by_valueref_no_var(self):
+        """Test get_variable_by_valueref for non-existing variable."""
+        fmu = self._get_reference_fmu("Feedthrough")
+        err_msg = "The variable with the valuref 10000 could not be found."
+        with pytest.raises(FMUException, match = re.escape(err_msg)):
+            fmu.get_variable_by_valueref(10000)
+
+    def test_get_variable_variability(self):
+        """Test get_variable_variability."""
+        fmu = self._get_reference_fmu("Feedthrough")
+        assert fmu.get_variable_variability("Float64_fixed_parameter") is FMI3_Variability.FIXED
+        assert fmu.get_variable_variability("Float64_tunable_parameter") is FMI3_Variability.TUNABLE
+        assert fmu.get_variable_variability("Float64_discrete_input") is FMI3_Variability.DISCRETE
+        assert fmu.get_variable_variability("Float64_continuous_input") is FMI3_Variability.CONTINUOUS
+
+    def test_get_variable_variability_no_var(self):
+        """Test get_variable_variability for non-existing variable."""
+        fmu = self._get_reference_fmu("Feedthrough")
+        err_msg = "The variable aaa could not be found."
+        with pytest.raises(FMUException, match = re.escape(err_msg)):
+            fmu.get_variable_variability("aaa")
+
+    def test_get_variable_initial(self):
+        """Test get_variable_initial."""
+        fmu = self._get_reference_fmu("Feedthrough")
+        assert fmu.get_variable_initial("Float64_continuous_input") is FMI3_Initial.EXACT
+        assert fmu.get_variable_initial("Float64_continuous_output") is FMI3_Initial.CALCULATED
+
+    def test_get_variable_initial_no_var(self):
+        """Test get_variable_initial for non-existing variable."""
+        fmu = self._get_reference_fmu("Feedthrough")
+        err_msg = "The variable aaa could not be found."
+        with pytest.raises(FMUException, match = re.escape(err_msg)):
+            fmu.get_variable_initial("aaa")
+
+    def test_get_variable_min(self):
+        """Test get_variable_min."""
+        fmu = FMUModelME3(str(this_dir / "files" / "FMUs" / "XML" / "ME3.0" / "variableAttributes"),
+                          allow_unzipped_fmu = True, _connect_dll = False)
+        assert fmu.get_variable_min("float64") == -6.4
+        assert fmu.get_variable_min("float32") == np.float32(-3.2)
```

Comment:
I guess this is a bigger question that is out of scope so asking mainly to learn/understand. Have we considered returning numpy objects instead of Python float? Now it seems like you get a Python float of 64-bit but the precision of it is actually worse.